crates/db_views/src/post_view.rs

```diff
@@ -160,8 +158,38 @@ impl PostView {
           post_aggregates::comments,
         ),
       ))
-      .first::<PostViewTuple>(conn)
-      .await?;
+      .into_boxed();
+
+    // Hide deleted and removed for non-admins or mods
+    if !is_mod_or_admin.unwrap_or(false) {
+      // If you are not the creator, then remove the other fields.
```

Comment:
Comment seems wrong, its excluding these posts from results not removing any fields.